### PR TITLE
added dependency for java 9 compatibility

### PR DESCRIPTION
**Pull request by jensonjoseph**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@jensonjoseph Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12167)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12167) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@jensonjoseph Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=12167)!

File: spring-boot-samples/spring-boot-sample-data-jpa/pom.xml [hunk 1]
```diff
@@ -30,12 +30,18 @@
 			<artifactId>h2</artifactId>
 			<scope>runtime</scope>
 		</dependency>
+		<dependency>
+			<groupId>javax.xml.bind</groupId>
+			<artifactId>jaxb-api</artifactId>
+			<version>2.3.0</version>
```
Comment (snicoll) [inline spring-boot-samples/spring-boot-sample-data-jpa/pom.xml]: We don't hardcode version in child pom.

Comment (jensonjoseph) [reply spring-boot-samples/spring-boot-sample-data-jpa/pom.xml]: Moved dependency version to parent pom.

File: spring-boot-project/spring-boot-dependencies/pom.xml [hunk 2]
```diff
@@ -1003,6 +1004,11 @@
 				<artifactId>validation-api</artifactId>
 				<version>${javax-validation.version}</version>
 			</dependency>
+			<dependency>
```
Comment (snicoll) [inline spring-boot-project/spring-boot-dependencies/pom.xml]: I don't know what version of `master` you have but the current master [already has this dependency management](https://github.com/spring-projects/spring-boot/blob/c27fa7bf91cd00aa12fad7dd6d60c2b64ac12247/spring-boot-project/spring-boot-dependencies/pom.xml#L1044)

**Comment by snicoll** [comment]
Thanks a lot for the PR but I think we need to fix this differently. I've created #12302

**Comment by jensonjoseph** [comment]
ok, thanks for reviewing!

**Pull request closed without merging**